addons/ailogistics/functions/fnc_logisticsMain.sqf

```diff
@@ -0,0 +1,110 @@
+#include "script_component.hpp"
+/*
+    Author:
+        Bridg
+
+    Description:
+        Main function
+
+    Parameters:
+
+
+    Return value:
+        Nothing
+*/
+
+
+if (!isServer) exitWith {};
+
+(_this select 1) params ["_module"];
+
+
+// get all modules parameters
+private _destinations = synchronizedObjects _module;
+private _time = _module getVariable [QGVAR(logisticsTimeout),0];
+private _groupSize = _module getVariable [QGVAR(picketGroupSize),0];
+private _picketUnits = _module getVariable [QGVAR(picketUnitPoolString),[]];
+private _sideFromModule = _module getVariable [QGVAR(side),0];
+
+// get side then update the variable with an actual side, get the updated version
+if (_sideFromModule == 0) then {
+    _module setVariable [QGVAR(side),EAST];
+} else {
+    _module setVariable [QGVAR(side),INDEPENDENT];
+};
+private _side = _module getVariable [QGVAR(side),EAST];
+
+
+
+// establish the route
+private _agent = createAgent ["C_Man_1", getPos _module, [], 0, "NONE"];
+private _car = "B_Quadbike_01_F" createVehicle (getPos _module);
+_agent moveInDriver _car;
+
+_agent setVariable [QGVAR(module),_module];
+
+// workaround ai spawning at the end the route due to double execution bug of pathCalculated EH.
+// if this is called twice by sync-ing more than one destination module, it only picks route. Seems that you have to wait until the path calc's once before you can do it again.
+// if there is a way to do the above, one start module could support multiple destinations 1, would also require multi vehicle support
+
+_agent setDestination [(_destinations select 0), "LEADER PLANNED", true];
+
+_agent addEventHandler ["PathCalculated", {
+    params ["_agent", "_path"];
+    private _module = _agent getVariable [QGVAR(module),objNull];
+    if ((_module getVariable [QGVAR(routePickets),false]) == false) exitWith {};
+    private _picketLocationsRaw = _module getVariable [QGVAR(picketLocations),[0.5]]; // "[0.4,0.7]"
+    private _picketLocationsParsed = parseSimpleArray _picketLocationsRaw;
+
+    // this prevents looping
+    deleteVehicle (vehicle _agent);
+    deleteVehicle _agent;
+
+    // workout the points in _path
+    private _picketPoints = [];
+    {
+        private _percent = (round(count _path) * _x);
+        _picketPoints pushBack (_path select _percent);
+    } forEach _picketLocationsParsed;
```

Comment:
Can be simplified, however the maths used is going to fail if edge numbers are used (Never trust the user to do what you intend).

e.g round (18 * 0.999) = round (17.982) = 18, can't select index 18 of a zero-indexed array with 18 elements (0-17).

Proably better to use `floor`: floor (18 * 0.999) = floor (17.982) = 17

```
private _pathCount = count _path;
private _picketPoints = _picketLocationsParsed apply {_path select (floor (_pathCount * _x))};
```